packages/flutter/lib/src/widgets/lazy_block.dart

```diff
@@ -87,27 +125,68 @@ class _LazyBlockState extends ScrollableState<LazyBlock> {
   Widget buildContent(BuildContext context) {
     return new LazyBlockViewport(
       startOffset: scrollOffset,
-      direction: config.scrollDirection,
+      mainAxis: config.scrollDirection,
       onExtentsChanged: _handleExtentsChanged,
       delegate: config.delegate
     );
   }
 }
 
+/// Signature used by [LazyBlockViewport] to report its interior and exterior dimensions.
 typedef void LazyBlockExtentsChangedCallback(double contentExtent, double containerExtent, double minScrollOffset);
 
+/// A viewport on an infinite list of variable height children.
+///
+/// [LazyBlockViewport] is a a general-purpose viewport for a large (or
+/// infinite) number of children that might not all have the same height. Rather
+/// than materializing all of its children, [LazyBlockViewport] asks its
+/// [delegate] to build child widgets lazily to fill itself. [LazyBlockViewport]
+/// caches the widgets it obtains from the delegate as long as they're visible.
+/// (See [LazyBlockDelegate.shouldRebuild] for details about how to evict the
+/// cache.)
+///
+/// [LazyBlockViewport] works by dead reckoning changes to its [startOffset]
+/// from the top of the first child that is visible in itself. For this reason,
+/// making large changes to the [startOffset] is expensive because
+/// [LazyBlockViewport] computes the size of every child between the old offset
+/// and the new offset.
+///
+/// Prefer [ListViewport] when all the children have the same height because
+/// it can use that property to be more efficient. Prefer [Viewport] with a
+/// single child.
```

Comment:
Maybe add "For a scrollable version of LazyBlockViewport, see LazyBlock" or some such.